moveos/moveos-stdlib/rooch-framework/sources/account.move

```diff
@@ -46,24 +99,213 @@ module rooch_framework::account{
+      let source_addr = signer::address_of(source);
+      let seed = generate_seed_bytes(&source_addr);
+      let resource_addr = create_resource_address(&source_addr, seed);
+      let resource_signer = if (exists_at(resource_addr)) {
+         let account = borrow_global<Account>(resource_addr);
+         assert!(account.sequence_number == 0, error::invalid_state(EResourceAccountAlreadyUsed));
+         create_signer(resource_addr)
+      } else {
+         create_account_unchecked(resource_addr)
+      };
+
+      // By default, only the SignerCapability should have control over the resource account and not the auth key.
+      // If the source account wants direct control via auth key, they would need to explicitly rotate the auth key
+      // of the resource account using the SignerCapability.
+      rotate_authentication_key_internal(&resource_signer, ZERO_AUTH_KEY);
+
+      assert!(!is_resource_account(resource_addr), error::invalid_state(EAccountIsAlreadyResourceAccount));
+      move_to(&resource_signer, ResourceAccount {});
+
+      let signer_cap = SignerCapability { addr: resource_addr };
+      (resource_signer, signer_cap)
+   }
+
+   /// This is a helper function to generate seed for resource address
+   fun generate_seed_bytes(addr: &address): vector<u8> acquires Account{
+      let sequence_number = Self::sequence_number(*addr);
+      // use rooch token balance as part of seed, just for new address more random.
+      // TODO token standar
+      // let balance = Self::balance<...>(*addr);
+
+      let seed_bytes = bcs::to_bytes(addr);
+      vector::append(&mut seed_bytes, bcs::to_bytes(&sequence_number));
+      // vector::append(&mut seed_bytes, bcs::to_bytes(&balance));
+
+      hash::sha3_256(seed_bytes)
+   }
+
+   /// This is a helper function to compute resource addresses. Computation of the address
+   /// involves the use of a cryptographic hash operation and should be use thoughtfully.
+   public fun create_resource_address(source: &address, seed: vector<u8>): address {
+      let bytes = bcs::to_bytes(source);
+      vector::append(&mut bytes, seed);
+      vector::push_back(&mut bytes, DERIVE_RESOURCE_ACCOUNT_SCHEME);
+      bcd::to_address(hash::sha3_256(bytes))
+   }
+
+   /// This function is used to rotate a resource account's authentication key to 0, so that no private key can control
+   /// the resource account.
+   public(friend) fun rotate_authentication_key_internal(account: &signer, new_auth_key: vector<u8>) acquires Account {
+      let addr = signer::address_of(account);
+      assert!(exists_at(addr), error::not_found(EAccountNotExist));
+      assert!(
+         vector::length(&new_auth_key) == AUTHENTICATION_KEY_LENGTH,
+         error::invalid_argument(EMalformedAuthenticationKey)
+      );
+      let account_resource = borrow_global_mut<Account>(addr);
+      account_resource.authentication_key = new_auth_key;
+   }
```

Comment:
Whether the `authentication_key` needs to be verified to be in a valid format？

Reply:
`authentication_key` is generated by public keys via sha3-256 hash, and length is 32 bytes, has no other format